### DISCUSSION TITLE
Upload from AsyncRead to s3

### DISCUSSION THREAD

**Discussion by snaggen** [awslabs/aws-sdk-rust#361] [Q&A] [answered]
I have an API where I receive an AsyncRead with a non-static lifetime, that I would like to upload to s3, is there any good and easy way to do this?  I cant find a way to create a byte stream from a Read, so currently I have an adapter that will provide a Stream, and then I can do 
```
let body: Body = Body::wrap_stream(stream);
let sdk_body = SdkBody::from(body);
let bytes_stream = ByteStream::new(sdk_body);
```
but that doesn't feel like the best way to do it, and I still get an error about Transfer encoding not being supported.

**Comment by snaggen** [comment, marked as answer]
To answer my own question, the answer is to not try to feed the reader in to the SDK somehow. Instead just read the reader and feed the result to a multipart upload.

**Reply by rcoh** [reply]
I think we should probably add an AsyncRead adapter though, I can imagine that that's a fairly common use case.
